src/v2/expressions.ts

```diff
@@ -1,54 +1,104 @@
+  test: Expression<boolean>;
+  ifTrue: T;
+  ifFalse: T;
 
-/**
- * Creates an Expression that will cause an Options field to take on a runtime-defined
- * value equal to a CF3 Param defined by the functions in v2/params.ts.
- */
-export function FromParam<T extends string | number | boolean>(
-  paramName: string
-): Expression<T> {
-  if (!/^[A-Z_][A-Z0-9_]*$/.test(paramName)) {
-    throw new Error(
-      `Param name ${paramName} must start with an uppercase ASCII letter or underscore` +
-        ', and then consist of uppercase ASCII letters, digits, and underscores.'
-    );
-  }
-  return `{{ params.${paramName} }}`;
+  constructor(test: Expression<boolean>, ifTrue: T, ifFalse: T) {
+    super();
+    this.test = test;
+    this.ifTrue = ifTrue;
+    this.ifFalse = ifFalse;
+  }
+
+  get value(): T {
+    return !!this.test.value ? this.ifTrue : this.ifFalse;
+  }
+
+  toString() {
+    return `${this.test} ? ${this.ifTrue} : ${this.ifFalse}`;
+  }
 }
 
-/**
- * Creates an Expression that will cause an Options field to take on a runtime-defined
- * value obtained by evaluating the provided CEL expression.
- * The CLI does not currently have the capability to evaluate arbitrary CEL expressions.
- * You almost certainly want to use FromParam() instead.
- */
-export function FromCEL<T extends string | number | boolean>(
-  celString: string
-): Expression<T> {
-  const numOpens = celString.split('{{').length - 1;
-  const numCloses = celString.split('}}').length - 1;
-  if (numOpens == 0 || numOpens !== numCloses) {
-    throw new Error(
-      `Provided CEL expression ${celString} doesn't seem to be well formed.`
-    );
-  }
-  return celString;
+export class CompareExpression<
+  T extends string | number | boolean | string[]
+> extends Expression<boolean> {
+  cmp: '==' | '>' | '>=' | '<' | '<=';
+  lhs: Expression<T>;
+  rhs: T;
+
+  constructor(cmp: '==' | '>' | '>=' | '<' | '<=', lhs: Expression<T>, rhs: T) {
+    super();
+    this.cmp = cmp;
+    this.lhs = lhs;
+    this.rhs = rhs;
+  }
+
+  get value(): boolean {
+    const left = this.lhs.value;
+    switch (this.cmp) {
+      case '==':
+        return left === this.rhs;
+      case '>':
+        return left > this.rhs;
+      case '>=':
+        return left >= this.rhs;
+      case '<':
+        return left < this.rhs;
+      case '<=':
+        return left <= this.rhs;
+    }
+  }
+
+  toString() {
+    return `${this.lhs} ${this.cmp} ${this.rhs}`;
+  }
+
+  then(ifTrue: T, ifFalse: T) {
+    return new TernaryExpression(this, ifTrue, ifFalse);
+  }
 }
 
 /**
- * Casts an Expression to its literal string value, for use by __getSpec()
- * @internal
+ * Creates an Expression that will cause an Options field to take on a runtime-defined
+ * value based on the value of a CF3 Param defined via the helpers in the v2/params
+ * namespace.
  */
-export function ExprString<Expression>(arg: Expression): string {
-  return (arg as unknown) as string;
+export class ParamExpression<
```

Comment:
I originally tried this, but I found returning an Expression<T> to be a dead end. For me, at least, I had to create new "Base" types for each expression so I could add the valid operators to them. So for example, a StringParam is a StringExpression, which implements `eq` in addition to being an Expression<string>.  The only time I gave up and returned Expression<T> is the `then` API, but really I should go back and modify that to have N overrides so that each return type is a FooExpression and not an Expression<Foo>.